src/types/index.ts

```diff
@@ -1,13 +1,27 @@
-export type Set = {
-  key: number;
+// Exercise
+export interface Exercise {
+  name: string;
+  items: ExerciseItem[];
+}
+
+export type ExerciseItem = Set | Rest | Note;
+
+export interface Set {
+  id: number;
   reps: string;
   rpe: string; // Rating of Perceived Exertion
   orm: string; // One Rep Max
-};
+}
 
-export interface Exercise {
-  name: string;
-  sets: Set[];
+export interface Rest {
+  id: number;
+  minutes: string;
+  seconds: string;
```

Comment:
hmm, how do we go about using string for minutes and seconds rather than type: numbers?  we might have to account for users inputting non numbers in there.  Or maybe im just tripping.  Maybe @thegunner686  should check this one as well

Reply:
actually nvm, I need the types haha, push it boss